pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -15,8 +15,10 @@ isExecutable() {
     # *or* there is an INTERP section. This also catches position-independent
     # executables, as they typically have an INTERP section but their ELF type
     # is DYN.
-    LANG=C readelf -h -l "$1" 2> /dev/null \
-        | grep -q '^ *Type: *EXEC\>\|^ *INTERP\>'
+    isExeResult="$(LANG=C readelf -h -l "$1" 2> /dev/null \
+        | grep '^ *Type: *EXEC\>\|^ *INTERP\>')"
+    # not using grep -q, because it can cause Broken pipe
+    [ -n "$isExeResult" ]
```

Comment:
What about using `grep -q -m1` instead? This should stop after a match has been found and not result in a broken pipe error.

Reply:
How can we verify this works? I can't reproduce original issue. The closest thing I could do is generate broken pipe from `pip list`:
```
$ (pip list | grep Package) 2>&1 | head
You are using pip version 18.1, however version 19.0.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Package                       Version                     

$ (pip list | grep -q Package) 2>&1 | head
Traceback (most recent call last):
  File "/nix/store/l95nkqp7bdimqnz9ixay1aahljzsz7vc-python-2.7.15/lib/python2.7/logging/__init__.py", line 892, in emit
    self.flush()
  File "/nix/store/l95nkqp7bdimqnz9ixay1aahljzsz7vc-python-2.7.15/lib/python2.7/logging/__init__.py", line 852, in flush
    self.stream.flush()
IOError: [Errno 32] Broken pipe
Logged from file list.py, line 234
Traceback (most recent call last):
  File "/nix/store/l95nkqp7bdimqnz9ixay1aahljzsz7vc-python-2.7.15/lib/python2.7/logging/__init__.py", line 892, in emit
    self.flush()

$ (pip list | grep -q -m1 Package) 2>&1 | head
Traceback (most recent call last):
  File "/nix/store/l95nkqp7bdimqnz9ixay1aahljzsz7vc-python-2.7.15/lib/python2.7/logging/__init__.py", line 892, in emit
    self.flush()
  File "/nix/store/l95nkqp7bdimqnz9ixay1aahljzsz7vc-python-2.7.15/lib/python2.7/logging/__init__.py", line 852, in flush
    self.stream.flush()
IOError: [Errno 32] Broken pipe
Logged from file list.py, line 234
Traceback (most recent call last):
  File "/nix/store/l95nkqp7bdimqnz9ixay1aahljzsz7vc-python-2.7.15/lib/python2.7/logging/__init__.py", line 892, in emit
    self.flush()
```